src/m365/spo/commands/file/file-retentionlabel-ensure.ts

```diff
@@ -92,7 +98,16 @@ class SpoFileRetentionLabelEnsureCommand extends SpoCommand {
 
   public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
     try {
-      const fileProperties = await this.getFileProperties(logger, args);
+      const fileProperties: FileProperties = await this.getFileProperties(logger, args);
+
+      const labelInformation: ListItemRetentionLabel = await spo.getWebRetentionLabelInformationByName(args.options.webUrl, args.options.name);
+
+      if (args.options.assetId && !labelInformation.isEventBasedTag) {
+        throw `The label that's being applied is not an event-based label`;
```

Comment:
Could you remove thise check? The AssetID field can just be updated, regardless whether the label in question is an event-based label.